test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,130 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+describe('Signature Verification', (): void => {
+  // Repeat test with different keys
+  for (let index = 0; index < 4; index++) {
+    it('verifies correct signature - EC Key', async (): Promise<void> => {
+      const keyPair = crypto.generateKeyPairSync('ec', {
+        namedCurve: 'secp256k1', // Allowed by FIDO spec
+        publicKeyEncoding: {
+          type: 'spki', // Key infrasructure
+          format: 'pem' // Encoding format
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem'
+        }
+      })
+
+      const challenge = 'Crypto Auth service Yay!'
+      const signer = crypto.createSign('SHA256')
+
+      signer.update(challenge)
+
+      const sign = signer.sign(keyPair.privateKey, 'base64')
+
+      expect(await verifySign(challenge, sign, keyPair.publicKey)).toEqual(true)
```

Comment:
I tend to prefer assertions to not contain the actual test logic. It's much more readable to keep them separate.

```suggestion
      const sign = signer.sign(keyPair.privateKey, 'base64')
      const result = verifySign(challenge, sign, keyPair.publicKey)

      expect(result).toEqual(true)
```

Also this is `async`, but I don't think it needs to be (see above)

Reply:
@ramanmangla does it need to be async at point of call? to just run in the background while other stuff is happening so as not to block